### PR TITLE
[Go] Extract original airdate

### PR DESCRIPTION
**Pull request by ischmidt20**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible.

Extracts airdate from Go streams (particularly ABC). Not sure how to properly safely get the first element of the list if the list does not exist, my solution feels a little hacky so feel free to update for style. May also need to update previously existing code to remove compat code to pass tests

### CODE REVIEW

File: yt_dlp/extractor/go.py [hunk 1]
```diff
@@ -316,4 +317,5 @@ def _real_extract(self, url):
             'thumbnails': thumbnails,
             'formats': formats,
             'subtitles': subtitles,
+            'timestamp': timeconvert(video_data.get('airdates', {}).get('airdate', [None])[0]),
```
Comment (pukkandan) [inline yt_dlp/extractor/go.py:320]:
```suggestion
            'timestamp': unified_timestamp(traverse_obj(video_data, ('airdates', 'airdate', 0))),
```

**Review by pukkandan — CHANGES_REQUESTED** [review]
add/fix tests

**Comment by ischmidt20** [comment]
Done. I removed one test because it was for a playlist, and I got an error (full file path redacted). Not sure how to properly format it

```
Traceback (most recent call last):
  File "...../yt-dlp/test/test_download.py", line 276, in test_template
    getattr(self, f'test_{name}_{i}' if i else f'test_{name}')()
  File "...../yt-dlp/test/test_download.py", line 184, in test_template
    self.assertTrue(res_dict['_type'] in ['playlist', 'multi_video'])
KeyError: '_type'
```

**Comment by pukkandan** [comment]
Something's broken then. I won't ask you to fix it, but add the test back. If we remove failing tests, then what's the point of tests?

**Comment by ischmidt20** [comment]
Sure, but in this case it looks like a problem with the testing mechanism, as opposed to the extractor itself, which is why I removed it

**Comment by pukkandan** [comment]
>  but in this case it looks like a problem with the testing mechanism

How so? From the error message you provided, it appears the extractor is returning a single video when the test definition say that a playlist is expected

**Comment by ischmidt20** [comment]
Ah yes, that was my mistake for not interpreting the test output correctly. I have added the test back (after modifying it to reflect the fact the previous URL no longer exists). Perhaps I will try to fix it, but I have not tried to code anything up for playlists yet so I'll probably just leave it. Sorry about that!

**Pull request merged**